netbox-prometheus-sd.py

```
@@ -6,88 +6,180 @@
-            if getattr(device, 'device_role', None):
-                labels['__meta_netbox_role'] = device.device_role.slug
-            if getattr(device, 'device_type', None):
-                labels['__meta_netbox_type'] = device.device_type.model
-            if getattr(device, 'rack', None):
-                labels['__meta_netbox_rack'] = device.rack.name
-            if getattr(device, 'site', None):
-                labels['__meta_netbox_pop'] = device.site.slug
-            if getattr(device, 'serial', None):
-                labels['__meta_netbox_serial'] = device.serial
-            if getattr(device, 'parent_device', None):
-                labels['__meta_netbox_parent'] = device.parent_device.name
-            if getattr(device, 'address', None):
-                labels['__meta_netbox_address'] = device.address
-            if getattr(device, 'description', None):
-                labels['__meta_netbox_description'] = device.description
-            try:
-                device_targets = json.loads(device.custom_fields[args.custom_field])
-            except ValueError:
-                continue  # Ignore errors while decoding the target json FIXME: logging
-
-            if not isinstance(device_targets, list):
-                device_targets = [device_targets]
-
-            for target in device_targets:
-                target_labels = labels.copy()
-                target_labels.update(target)
-                if hasattr(device, 'primary_ip'):
-                    address = device.primary_ip
+class Discovery(object):
+    def __init__(self, args):
+        super(Discovery, self).__init__()
+        self.args = args
+
+    def run(self):
+        self.netbox = pynetbox.api(self.args.url, token=self.args.token)
+        if self.args.discovery == 'device':
+            targets = self.discover_device()
+        elif self.args.discovery == 'circuit':
+            targets = self.discover_circuit()
+        else:
+            return
```

Should raise an error if the discovery method is unknown